src/object.js

```diff
@@ -41,28 +41,49 @@ function TransisObject() {
 
 TransisObject.displayName = 'Transis.Object';
 
-// Internal: Processes a property change by traversing the property dependency graph and forwarding
-// changes to proxy objects.
-function didChange(object, name) {
-  (object.__changedProps__ = object.__changedProps__ || {})[name] = true;
-  changedObjects[object.objectId] = object;
-  uncache.call(object, name);
+// Internal: Processes recorded property changes by traversing the property dependency graph and
+// forwarding changes to proxy objects.
+function processChangedObjects() {
+  let seen = {};
+  let head;
 
-  if (object.__deps__) {
-    let deps = object.__deps__[name];
+  for (let id in changedObjects) {
+    for (let k in changedObjects[id].__changedProps__) {
+      head = {object: changedObjects[id], name: k, next: head};
+      seen[changedObjects[id].objectId + k] = true;
+    }
+  }
+
+  while (head) {
+    let object = head.object;
```

Comment:
Maybe to DRY-up the code below and reduce property lookups (ex. `object.objectId`), we could destructure upfront.

```js
let {name, object, object:{objectId, __deps__, __proxies__}} = head
```